cumulusci/salesforce_api/metadata.py

```diff
@@ -562,10 +560,16 @@ def _process_response(self, response):
             for tag in tags:
                 result_data[tag] = self._get_element_value(result, tag)
             # Parse dates
-            # FIXME: This was breaking things
-            # for key in parse_dates:
-            #    if result_data[key]:
-            #        result_data[key] = dateutil.parser.parse(result_data[key])
+            for key in parse_dates:
+                if result_data[key]:
+                    try:
+                        result_data[key] = parse_api_datetime(result_data[key])
+                    except Exception:
+                        # parsing the datetime with the dateutil library was "causing problems"
+                        # according to an old comment without much detail, so i'm being
+                        # unnecessarily defensive here.
+                        # TODO: determine if this is needed.
+                        pass
```

Comment:
I think I would lean toward not being defensive, but raising an exception including the value that couldn't be parsed. That way we'll actually find out what kind of values we aren't handling properly rather than silently pretending that data doesn't exist.